### PR TITLE
more bug fixing and improvement

### PR DESCRIPTION
**Pull request by jereolivieri**

### CODE REVIEW

File: src/lib/apis/autoptic/index.ts [hunk 1]
```diff
@@ -518,28 +518,32 @@ export const getListPQL = async () => {
...
-		method: 'GET',
-		headers: {
-			'Content-Type': 'application/json',
-			// Authorization: `Bearer ${token}`
-		}
-	})
-		.then(async (res) => {
-			if (!res.ok) throw await res.json();
-			return res.json();
+	if (endpoint_id !== '' && serverURL !== '') {
```
Comment (fdiazaguirre) [inline src/lib/apis/autoptic/index.ts:521]: @jereolivieri deberías hacer una validación más sofisticada utilizando una RegEx. 
```suggestion
	if (typeof endpoint_id === 'string' && endpoint_id.trim() !== '' &&
    typeof serverURL === 'string' && serverURL.trim() !== '') {
    // Your code here
}

function isValidServerUrl(url) { const serverUrlRegex = /^(https?:\/\/)?([a-zA-Z0-9\-\.]+)(\.[a-zA-Z]{2,})(:[0-9]{1,5})?(\/.*)?$/; return serverUrlRegex.test(url); }
```

File: src/lib/apis/autoptic/index.ts [hunk 2]
```diff
@@ -518,28 +518,32 @@ export const getListPQL = async () => {
...
-	if (error) {
-		throw error;
+			.then(async (res) => {
+				if (!res.ok) throw await res.json();
+				return res.json();
+			})
+			.catch((err) => {
+				console.log(err);
+				error = err.detail;
+				return null;
```
Comment (fdiazaguirre) [inline src/lib/apis/autoptic/index.ts:536]: re-throw the original exception, and stop with the overriding [behavior anti-pattern](https://stackoverflow.com/questions/24623484/exception-handling-anti-patterns-why-log-and-return-null-is-an-anti-pattern)!

File: src/lib/apis/autoptic/index.ts [hunk 3]
```diff
@@ -518,28 +518,32 @@ export const getListPQL = async () => {
...
+			});
+		if (error) {
+			throw error;
+		}
+		
+		return res;
+	} else {
+		return []
```
Comment (fdiazaguirre) [inline src/lib/apis/autoptic/index.ts:544]: if an empty array is the default value, initialize the res variable with that and reduce the [cyclomatic complexity](https://www.geeksforgeeks.org/cyclomatic-complexity/#)!

File: src/lib/apis/autoptic/index.ts [hunk 4]
```diff
@@ -580,27 +584,32 @@ export const getDefaultListSnapshots = async (window: string) => {
...
-		method: 'GET',
-		headers: {
-			'Content-Type': 'application/json',
-			// Authorization: `Bearer ${token}`
-		}
-	})
-		.then(async (res) => {
-			if (!res.ok) throw await res.json();
-			return res.json();
+	if (endpoint_id !== '' && serverURL !== '') {
```
Comment (fdiazaguirre) [inline src/lib/apis/autoptic/index.ts:587]: ditto

File: src/lib/apis/autoptic/index.ts [hunk 5]
```diff
@@ -580,27 +584,32 @@ export const getDefaultListSnapshots = async (window: string) => {
...
-	if (error) {
-		throw error;
+			.then(async (res) => {
+				if (!res.ok) throw await res.json();
+				return res.json();
+			})
+			.catch((err) => {
+				console.log(err);
+				error = err.detail;
+				return null;
```
Comment (fdiazaguirre) [inline src/lib/apis/autoptic/index.ts:602]: ditto

File: src/lib/apis/autoptic/index.ts [hunk 6]
```diff
@@ -580,27 +584,32 @@ export const getDefaultListSnapshots = async (window: string) => {
...
+			});
+		if (error) {
+			throw error;
+		}
+
+		return res;
+	} else {
+		return []
```
Comment (fdiazaguirre) [inline src/lib/apis/autoptic/index.ts:610]: ditto

**Review by fdiazaguirre — CHANGES_REQUESTED** [review]
Needs more work!

File: src/lib/apis/autoptic/index.ts (hunk 1 above)
Comment (jereolivieri) [reply src/lib/apis/autoptic/index.ts:521]: The formatting and validation of the URL was done in another part of the application. So, is not necessary here.

Comment (jereolivieri) [reply src/lib/apis/autoptic/index.ts:521]: it’s not exactly the same regex validation, and due to how the code was previously written, it needs to be done this way

**Review by fdiazaguirre — APPROVED** [review]
LGTM

**Pull request merged**